src/actions/secrets.js

```diff
@@ -136,34 +138,56 @@ export function deleteSecret(secrets, cancelMethod) {
 /* istanbul ignore next */
 export function createSecret(postData, namespace) {
   return async dispatch => {
+    dispatch({
+      type: 'CLEAR_SECRET_ERROR_NOTIFICATION'
+    });
     dispatch({ type: 'SECRET_CREATE_REQUEST' });
     try {
       await createCredential(postData, namespace);
-      const serviceAccount = await getServiceAccount({
-        name: postData.metadata.labels.serviceAccount,
-        namespace
-      });
-      await patchServiceAccount({
-        serviceAccountName: serviceAccount.metadata.name,
-        namespace: serviceAccount.metadata.namespace,
-        secretName: postData.metadata.name
-      });
-      dispatch({
-        type: 'CLEAR_SECRET_ERROR_NOTIFICATION'
-      });
       dispatch({
         type: 'SECRET_CREATE_SUCCESS'
       });
-      return false;
     } catch (error) {
       error.response.text().then(message => {
         dispatch({ type: 'SECRET_CREATE_FAILURE', error: message });
       });
-      return true;
     }
   };
 }
 
+export function patchSecret(serviceAccounts, secret, handleClose) {
+  return async dispatch => {
+    dispatch({
+      type: 'CLEAR_SECRET_ERROR_NOTIFICATION'
+    });
+    dispatch({ type: 'SECRET_PATCH_REQUEST' });
+    return Promise.all(
+      serviceAccounts.map(serviceAccount => {
+        return patchServiceAccount({
+          serviceAccountName: serviceAccount.name,
+          namespace: serviceAccount.namespace,
+          secretName: secret
+        });
+      })
+    )
+      .then(() => {
+        dispatch({
+          type: 'SECRET_PATCH_SUCCESS'
+        });
+        handleClose();
+      })
+      .catch(error => {
+        error.response.text().then(message => {
+          dispatch({ type: 'SECRET_PATCH_FAILURE', error: message });
+        });
+      });
+  };
+}
+
 export function clearNotification() {
   return { type: 'CLEAR_SECRET_ERROR_NOTIFICATION' };
 }
+
+export function resetCreate() {
+  return { type: 'RESET_CREATE' };
+}
```

Comment:
```suggestion
export function resetCreateSecret() {
  return { type: 'RESET_CREATE_SECRET };
}
```

Include reference to 'secret' in the naming so it's clear when used.